### DISCUSSION TITLE
android PagImageView本地缓存疑惑

### DISCUSSION THREAD

**Discussion by zengke328** [Tencent/libpag#2157] [General] [answered]
使用官方demo运行后【https://github.com/libpag/pag-android 】， 本地会生成很多缓存文件，怀疑是pag磁盘缓存文件，刚好对应是22个【见图  pag本地缓存.png】。但是看源码没有看到生成缓存的地方。
我在pag官网介绍上说是【PAGImageView 默认是会开启磁盘缓存的，但是对于内存缓存考虑内存限制，默认是关闭的。https://pag.art/docs/use-pagimageview.html 】

疑惑点1：本地缓存的文件是怎么生成的。
疑惑点2：本地缓存文件是否生效，我看源码是直接加载的assets文件（ReadBytesFromAssets）。【https://github.com/Tencent/libpag/blob/main/src/platform/android/JNIHelper.cpp#L104】

![pag2](https://github.com/Tencent/libpag/assets/22447723/4f046d18-8368-4e71-ac5e-b3af45d63745)
![pag3](https://github.com/Tencent/libpag/assets/22447723/8f210c01-8db6-428f-a528-a4404b95abb6)
![pag本地缓存](https://github.com/Tencent/libpag/assets/22447723/1ee5e0c6-c17c-44ef-9c87-eb0e65b8cc67)
![pag加载assets](https://github.com/Tencent/libpag/assets/22447723/72111889-1972-45c1-84fe-1a57b0764667)

**Comment by kevingpqi123** [comment, marked as answer]
1、对于 PAGImageView 而言，一个 pag 文件对应一个缓存文件，如果加载了多个 pag 文件，就会有多个缓存文件，缓存文件和 PAGImageView 传入的 path 强相关
2、pag 是跨平台的，核心的代码实现都在 C++ 层，平台层仅仅接口封装和桥接，因此磁盘缓存相关功能在 C++ 层的代码，而不是平台层代码，如果想了解，可以看下 https://github.com/Tencent/libpag/blob/main/src/rendering/caches/DiskCache.cpp
caches 文件夹下面的相关代码
